src/liseq.py

```diff
@@ -31,4 +31,18 @@ def liseq(x: Sequence[Any]) -> list[int]:
     """
     best: list[int] = []
     # Explore all alternatives
+    for i in range(len(x)):
```

Comment:
This is fine, but you don't need to increase `i` by one every time (as you implicitly do here). When you have made it through an increasing substring, you will end up at the same location every time you start inside it again. You could just jump to the end of that substring instead.